### ISSUE TITLE
CSV product import does not format url key when `url_key` column is used

### ISSUE THREAD

**Issue by rikwillems** [magento/magento2#32911]
When I use the Magento default csv product import with an `url_key` field the value is not formatted/translit before being inserted into the database. This way I end up with spaces in the url key field. The rest of the system works ok with this except GraphQL which will output the `url_key` field directly from the database.

### Preconditions (*)
Magento 2.4.2 and probably all versions below.

### Steps to reproduce (*)
1. Create a product with sku: ABC
2. Import csv below:
```
"sku", "url_key"
"ABC", "Test ABC 1 2 3"
```

### Expected result (*)
1. `url_key` attribute being: `test-abc-1-2-3`

### Actual result (*)
1. `url_key` attribute being: `test abc 1 2 3`

---
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._

## Patch

```
--- vendor/magento/module-catalog-import-export/Model/Import/Product.php
+++ vendor/magento/module-catalog-import-export/Model/Import/Product.php
@@ -3011,7 +3011,7 @@
     {
         if (!empty($rowData[self::URL_KEY])) {
             $urlKey = (string) $rowData[self::URL_KEY];
-            return trim(strtolower($urlKey));
+            return $this->productUrl->formatUrlKey($urlKey);
         }
 
         if (!empty($rowData[self::COL_NAME])
```

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Oscar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-42145` were created

**Issue Available**: @engcom-Oscar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by SilinMykola** [comment]
@magento I am working on this

**Comment by gabrieldagama** [comment]
Hi @rikwillems. Thank you for your report. 
The issue has been fixed in magento/magento2#33088 by @SilinMykola in 2.4-develop branch
Related commit(s):
 - [a7b9176ac292479705047f63c6d80ce6c6e55f92](https://github.com/magento/magento2/commit/a7b9176ac292479705047f63c6d80ce6c6e55f92)

The fix will be available with the upcoming 2.4.3 release.
